lib/rubocop/cop/style/pattern_argument.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop looks for all? / any? / none? / one? calls where the block
+      # has a single statement and === has been invoked on the receiver so the
+      # === argument can be used as the predicate method argument.
+      #
+      # @example
+      #   # bad
+      #   %w[foo bar foobar].any? { |e| /oo/.match?(e) }
+      #
+      #   [1, 2, 3].any? { |e| (1..10).include?(e) }
+      #
+      #   [1, 2, 3].any? { |e| Set[1, 2, 3] === e }
+      #
+      #   [1, 2, 3].none? do |e|
+      #     Integer === e
+      #   end
+      #
+      #   # good
+      #   %w[foo bar foobar].any?(/oo/)
+      #
+      #   [1, 2, 3].any?(1..10)
+      #
+      #   [1, 2, 3].any?(Set[1, 2, 3])
+      #
+      #   [1, 2, 3].none?(Integer)
+      #
+      class PatternArgument < Cop
+        include RangeHelp
+        extend TargetRubyVersion
+
+        minimum_target_ruby_version 2.5
+
+        def on_block(node)
+          pattern_argument_candidate?(node) do |method, pattern|
+            add_offense(node,
+                        location: method.loc.selector,
+                        message: format(MSG, method: method.method_name, pattern: pattern.source))
+          end
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            _, replacement = pattern_argument_candidate?(node).map(&:source)
+            replacement = "(#{replacement})" unless replacement.match?(/^\(.+\)$/)
+            corrector.replace(block_range_with_space(node), replacement)
+          end
+        end
+
+        private
+
+        MSG = 'Pass `%<pattern>s` as an argument to `%<method>s` instead of a block.'
+
+        def_node_matcher :pattern_argument_candidate?, <<~PATTERN
+          (block $({send csend} _ {:all? :any? :none? :one?}) (:args _) {
+            (send ${(begin ({irange erange} _ _))
+                    (send (const nil? :Range) :new _ _)
```

Comment:
`const {nil? cbase}` is better. We'll soon have `global_const?(:Range)` which will be even better.